src/thor/route_matcher.cc

```diff
@@ -277,7 +319,7 @@ bool RouteMatcher::FormPath(const std::shared_ptr<DynamicCost>* mode_costing,
     // Process directed edge and info
     const DirectedEdge* de = begin_edge_tile->directededge(graphid);
     const GraphTile* end_node_tile = reader.GetGraphTile(de->endnode());
-    if (begin_edge_tile == nullptr) {
+    if (end_node_tile == nullptr) {
```

Comment:
what is the rule of nullptr checking? Maybe it is better to have consistency by enforce compare with nullptr?